core/validator/src/rules/duplicate-asset-name.ts

```diff
@@ -50,16 +51,18 @@ export class DuplicateAssetName extends BaseValidator {
     validations: Record<string, StateOutput>
   ): Record<string, StateOutput> {
     const errorsMetadata = new Set<Metadata>();
+    const assetNameCount: Record<string, number> = {};
 
+    // First pass: Count occurrences of each assetName
     for (const entry of metadatas) {
-      const founds = metadatas.filter(
-        (meta) => meta.assetName === entry.assetName
-      );
+      assetNameCount[entry.assetName] =
```

Comment:
cant we build the Map or Set directly in that ? 
like updating the AsserNamecount (as a Set) then check the quantity and if > 1 update the error on the fly ?